From e328a3c25d7e74c98e3c1a1271a87cfe59b79a27 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 22 Jan 2024 20:32:41 +0100
Subject: [PATCH 547/954] docs(dt-bindings): add DDR supply in STM32MP2 device
 tree bindings

Add supply name in the DDR binding documentation file for STM32MP2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 08b0293e7d2c9da5f7f76ab3ff42438b8b5e4b73 docs(dt-bindings): add DDR supply in STM32MP2 device tree bindings]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Id3abcf5e44d6326ac369e2a8a153cd7f0112774f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/355236
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/368531
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 .../memory-controllers/st,stm32mp2-ddr.txt     | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/docs/devicetree/bindings/memory-controllers/st,stm32mp2-ddr.txt b/docs/devicetree/bindings/memory-controllers/st,stm32mp2-ddr.txt
index 75533a1745..6b79a7e32e 100644
--- a/docs/devicetree/bindings/memory-controllers/st,stm32mp2-ddr.txt
+++ b/docs/devicetree/bindings/memory-controllers/st,stm32mp2-ddr.txt
@@ -6,6 +6,24 @@ Required properties:
 - compatible	: Should be "st,stm32mp2-ddr"
 - reg		: controller (DDRCTRL) and phy (DDRPHYC) base address
 
+And the supply attributes are depending on the DDR type (DDR3/DDR4/LPDDR4).
+
+For DDR3:
+- vdd-supply	: phandle to the VDD/VDDQ regulator device tree node.
+- vref-supply	: phandle to the VREFCA/VREFDQ regulator device tree node.
+- vtt-supply	: phandle to the VTT regulator device tree node.
+
+For DDR4:
+- vdd-supply	: phandle to the VDD/VDDQ regulator device tree node.
+- vref-supply	: phandle to the VREFCA regulator device tree node.
+- vpp-supply	: phandle to the VPP regulator device tree node.
+- vtt-supply	: phandle to the VTT regulator device tree node.
+
+For LPDDR4:
+- vdd1-supply	: phandle to the VDD1 regulator device tree node.
+- vdd2-supply	: phandle to the VDD2/VDDCA regulator device tree node.
+- vddq-supply	: phandle to the VDDQ regulator device tree node.
+
 the next attributes are DDR parameters, they are generated by DDR tools
 included in STM32 Cube tool
 
-- 
2.53.0

